tools/shaderc/shaderc_metal.cpp

```diff
@@ -266,10 +266,113 @@ namespace bgfx { namespace metal
 		return size;
 	}
 
-	static bool compile(const Options& _options, uint32_t _version, const std::string& _code, bx::WriterI* _shaderWriter, bx::WriterI* _messageWriter, bool _firstPass)
+	static spv_target_env getSpirvTargetVersion(uint32_t version)
 	{
-		BX_UNUSED(_version);
+		switch (version)
+		{
+		case 1000:
+		case 1110:
+		case 1210:
+			return SPV_ENV_VULKAN_1_0;
+		case 2011:
+		case 2111:
+		case 2211:
+			return SPV_ENV_VULKAN_1_1;
+		case 2314:
+		case 2414:
+		case 3014:
+		case 3114:
+			return SPV_ENV_VULKAN_1_1_SPIRV_1_4;
+		default:
+			BX_ASSERT(0, "Unknown SPIR-V version requested. Returning SPV_ENV_VULKAN_1_0 as default.");
+			return SPV_ENV_VULKAN_1_0;
+		}
+	}
+
+	static glslang::EShTargetLanguageVersion getGlslangTargetSpirvVersion(uint32_t version)
+	{
+		switch (version)
+		{
+		case 1000:
+		case 1110:
+		case 1210:
+			return glslang::EShTargetSpv_1_0;
+		case 2011:
+		case 2111:
+		case 2211:
+			return glslang::EShTargetSpv_1_1;
+		case 2314:
+		case 2414:
+		case 3014:
+		case 3114:
+			return glslang::EShTargetSpv_1_4;
+		default:
+			BX_ASSERT(0, "Unknown SPIR-V version requested. Returning EShTargetSpv_1_0 as default.");
+			return glslang::EShTargetSpv_1_0;
+		}
+	}
+
+	static spirv_cross::CompilerMSL::Options::Platform getMslPlatform(const std::string &platform)
+	{
+		return "ios" == platform
+			? spirv_cross::CompilerMSL::Options::Platform::iOS
+			: spirv_cross::CompilerMSL::Options::Platform::macOS;
+	}
+
+	static void getMSLVersion(uint32_t version, uint32_t& major, uint32_t& minor)
+	{
+		switch (version)
+		{
+		case 1000:
+			major = 1;
+			minor = 0;
+			return;
+		case 1110:
+			major = 1;
+			minor = 1;
+			return;
+		case 1210:
+			major = 1;
+			minor = 2;
+			return;
+		case 2011:
+			major = 2;
+			minor = 0;
+			return;
+		case 2111:
+			major = 2;
+			minor = 1;
+			return;
+		case 2211:
+			major = 2;
+			minor = 2;
+			return;
+		case 2314:
+			major = 2;
+			minor = 3;
+			return;
+		case 2414:
+			major = 2;
+			minor = 4;
+			return;
+		case 3014:
+			major = 3;
+			minor = 0;
+			return;
+		case 3114:
+			major = 3;
+			minor = 1;
+			return;
+		default:
+			BX_ASSERT(0, "Unknown MSL version requested. Returning 1.0 as default.");
```

Comment:
Should remove assert, because this is tool. Should be warning/info into output that always fire in case of downgrade.